trillian/ctfe/config.go

```diff
@@ -210,6 +214,24 @@ func ValidateLogConfig(cfg *configpb.LogConfig) (*ValidatedLogConfig, error) {
 		}
 	}
 
+	// Validate CTFEStorageConnectionString
+	if len(cfg.CtfeStorageConnectionString) > 0 {
+		if strings.HasPrefix(cfg.CtfeStorageConnectionString, "mysql") {
+			if _, err := mysql.ParseDSN(strings.Split(cfg.CtfeStorageConnectionString, "://")[1]); err != nil {
+				return nil, errors.New("failed to parse ctfe_storage_connection_string for mysql driver")
+			}
+		} else {
+			return nil, errors.New("unsupported driver in ctfe_storage_connection_string")
+		}
+		vCfg.CTFEStorageConnectionString = cfg.CtfeStorageConnectionString
+	}
+
+	// Validate the combination of CtfeStorageConnectionString and ExtraDataIssuanceChainStorageBackend
+	if len(cfg.CtfeStorageConnectionString) == 0 && cfg.ExtraDataIssuanceChainStorageBackend == configpb.LogConfig_ISSUANCE_CHAIN_STORAGE_BACKEND_CTFE {
```

Comment:
I wonder if might might be more natural to rearrange this to be something like:

```
switch cfg.ExtraDataIssuanceChainStorageBackend {
case configpb.LogConfig_ISSUANCE_CHAIN_STORAGE_BACKEND_CTFE:
  if len(cnxnString) == 0 { return error }
  // validate cnxnString is sensible and return errors otherwise
case configpb.LogConfig_ISSUANCE_CHAIN_STORAGE_BACKEND_TRILLIAN:
  // whatever
}
```

That way validation is more closely grouped with the naming of the configured storage, and validation is only done when the string will actually be used - I think this might help improving readability a bit?